src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/ItemManager.cs

```diff
@@ -290,7 +290,10 @@ private static bool IsAfterDot(ITextSnapshot snapshot, ITrackingSpan applicableT
             if (bestItem != null)
             {
                 selectedItemIndex = itemsInList.IndexOf(i => Equals(i.FilterResult.CompletionItem, bestItem));
-                if (selectedItemIndex > -1 && bestItem != null && matchingItems.Length == 1 && filterText.Length > 0)
+                if (selectedItemIndex > -1 &&
+                    bestItem != null &&
+                    matchingItems.Where(r => !r.DisplayText.StartsWith("★")).Count() == 1 &&
```

Comment:
What would this look like if we did it in a way that didn't hard-code specific implementation details of Intellicode?